### ISSUE TITLE
torch.max gives different results on CPU and CUDA

### ISSUE THREAD

**Issue by bonlime** [pytorch/pytorch#37921]
## 🐛 Bug

The result of `torch.max` with `dim` is different on CPU and on CUDA. CPU version return the last element, while CUDA version returns first.  
 
## To Reproduce

```
In [1]: import torch                                                                                                                                  

In [2]: torch.max(torch.ones(1, 4), dim=1)                                                                                                                
Out[2]: 
torch.return_types.max(
values=tensor([1.]),
indices=tensor([3]))

In [3]: torch.max(torch.ones(1, 4).cuda(), dim=1)                                                                                                         
Out[3]: 
torch.return_types.max(
values=tensor([1.], device='cuda:0'),
indices=tensor([0], device='cuda:0'))
```

## Expected behavior

CPU and CUDA version should give the same result for sure.  

In my opinion, `cuda` behavior is correct because it aligns with NumPy. 
From the other hand current `torch.argmax` implementation follows the cpu behaviour. 

upd. There is an open pull request that fixes this behavior for argmax #37864 . 

## Environment
```
In [4]: torch.__version__                                                                                                                             
Out[4]: '1.5.0a0+8f84ded'
```

**Comment by ngimel** [comment]
This is a duplicate of many issues (see e.g. #17738 and issues linked there). This behavior is expected and documented https://pytorch.org/docs/master/generated/torch.max.html#torch.max, closing.
